src/main/java/uk/gov/hmcts/divorce/solicitor/service/SolicitorSubmitApplicationService.java

```diff
@@ -83,9 +85,18 @@ public OrderSummary getOrderSummary() {
         applicantSubmittedNotification.send(caseData, caseId);
         solicitorSubmittedNotification.send(caseData, caseId);
 
+        State state = SolicitorAwaitingPaymentConfirmation;
+
+        List<String> submittedErrors = Submitted.validate(caseData);
+        if (submittedErrors.isEmpty()) {
+            caseData.setDateSubmitted(LocalDateTime.now());
```

Comment:
You could use an Autowired Clock, and use that to create the LocalDateTime, e.g. LocalDateTime.now(clock).  You can then unit test this easily.
See ClockConfiguration class for the default autowired clock instance.

Reply:
Will take a look, thanks